.travis.yml

```diff
@@ -17,23 +18,38 @@ env:
     -  secure: "wRUxg3L05UjigEq0zPTfW+vFEPdDQrWFB8O3/p/7LYYL1EqnIvQETd/4g5vnzyu4iysPeCHtMY7a+oyX/LM81JD3EU2nzxfY1bRDL6kyU+KrWNhZ6Jf4XosCUQ1vpdD0jBIlpLoTLr056PUNrS/13onnJHVAWzIhjcDwUJXQlyzR8z32SN3Wa6yjQhHwGvQqqdJB/H83H0BcYTCxwKfut58zQZ+RVcubHj6T+K3LI8OPHvZeK2/NiWcaZB/zky5cD9lFiY7lF+/0PhYPw1EYwbzA2dPhyKxy+d8hweCEFMUb5Z9s980tAd8m1IdqGXPZu51lvj7MMWwj4X2vUUtF9XrB4K2Kt11PSyB49bF2brQI2axsyYWZGeW0nsb8+MlHUa3eli7XR6dA6VVeIJT08ULvXUVzEvXiogHE/5G9ZXVFWDgCkFsXdEAtf2EOo3RpCoUYSXmqRvr9ZBP6C/KMYogENQ8JdTk3SXpn8heo6oraBKfOg0DRfoiMVyl7vfOVQYRgbbDGZVJfcHSYpyFQfqKOMlEiSSyh51PL7Dt6BYSCBoUKh8AAAtIi2+exgTuUHwy8fZBBJm5Bbem8Ku9il89gkOWS8jPYxat8/9HlouNO/COBBb3DUPIKEOE+YwLji44GiVOB4bnZ1Hz1ucBZEvYzFSW3/wac7S9fqDaeOVE="
 
   matrix:
-  - NODE_VER=8 FULL_VALIDATE=true alias grunt=./node_modules/grunt-cli/bin/grunt
-  - NODE_VER=9 FULL_VALIDATE=false
+  - NODE_VER=8 FULL_VALIDATE=true
+  - NODE_VER=8 TSC_VERSION=latest
+  - NODE_VER=8 TSC_VERSION=next
+  - NODE_VER=9
+  - NODE_VER=9 TSC_VERSION=latest
+  - NODE_VER=9 TSC_VERSION=next
+  - NODE_VER=10
+  - NODE_VER=10 TSC_VERSION=latest
+  - NODE_VER=10 TSC_VERSION=next
 
 matrix:
   fast_finish: true
+  allow_failures:
+    - NODE_VER=8 TSC_VERSION=latest
+    - NODE_VER=8 TSC_VERSION=next
+    - NODE_VER=9 TSC_VERSION=latest
+    - NODE_VER=9 TSC_VERSION=next
+    - NODE_VER=10 TSC_VERSION=latest
+    - NODE_VER=10 TSC_VERSION=next
 
 before_install:
   - nvm install $NODE_VER
-  - npm install -g npm@6 && npm install -g npx && node -v && npm -v
+  - npm install -g npm@6 && node -v && npm -v
   - npm install -g yarn
-  - if [ "$FULL_VALIDATE" == "true" ]; then npm install --no-save grunt@0.4.1 grunt-cli grunt-contrib-connect grunt-run; fi
 
 install:
   - npm ci
   - if [ "$FULL_VALIDATE" == "true" ]; then npm run lint && npm run test:circular; fi
+  - if [ "${TSC_VERSION}" ]; then npm install --no-save typescript@$TSC_VERSION && ./node_modules/.bin/tsc --version && npm run build_cjs ; fi
 
 script:
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
```

Comment:
`$TRAVIS_BRANCH` points head of PR (most cases `master`), so https://github.com/ReactiveX/rxjs/pull/3771/files#diff-354f30a63fb0907d4ad57269548329e3L46 tends to be triggered for each PR.